### PR TITLE
Media: Avoid taking an existing element's base color

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

While working on another ticket, I noticed that sometimes, the base color was repeated the same color. This was becuase it was reusing the existing element on the page. By passing a unique id, it creates a new element and fixes the issue. 

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [ ] I have added a matching `Type: XYZ` label to the PR

---

Fixes #

### CODE REVIEW

**Comment by swissspidy** [comment]
This makes me wonder: Why are the img elements added to the DOM in the first place? Seems unnecessary. Avoiding that would fix this too.

**Comment by spacedmonkey** [comment]
> This makes me wonder: Why are the img elements added to the DOM in the first place? Seems unnecessary. Avoiding that would fix this too.

@miina I wonder this as well. Can you explain why this is needed?

**Comment by miina** [comment]
> @miina I wonder this as well. Can you explain why this is needed?

Perhaps the base color was previously created based on some URLs of images when they weren't anywhere yet?
I don't remember why it was added like this, by all means, feel free to change it! The `baseColor` logic has been completely reworked meanwhile, we should def opt for the most suitable approach.

**Comment by swissspidy** [comment]
I‘d recommend opening an accompanying ticket if this requires more work.

**Comment by spacedmonkey** [comment]
I now just create a unquie id for every base color generation. I will create a ticket for refactoring the logic. It is a little hard to understand as it. Not sure I understand why we create a div, we can just pass in a node.

**Comment by swissspidy** [comment]
If we're gonna fix it, let's fix it properly in the new year

File: packages/story-editor/src/utils/getMediaBaseColor.js [hunk 1]
```diff
@@ -93,17 +94,16 @@ export function setOrCreateImage(
   imageData,
   getOnloadCallback = getDefaultOnloadCallback
 ) {
-  const { src, id, height, width } = imageData;
+  const { src, height, width } = imageData;
+  const id = uuidv4();
   return new Promise((resolve, reject) => {
     const NODE_KEY = getImgNodeKey(id);
-    let imgNode = document.body[NODE_KEY];
```
Comment (miina) [inline packages/story-editor/src/utils/getMediaBaseColor.js]: One reason for re-using the same image node was not to create a bunch of nodes hanging around in the DOM but just reusing the same image tag. Why is adding a new node every time better than using the same one node?

If I understood correctly, Pascal's question was generally about adding the node to the DOM, this still does it, no?

**Comment by miina** [comment]
> It is a little hard to understand as it. Not sure I understand why we create a div, we can just pass in a node.

@spacedmonkey the `div` was previously created _because_ we added the node to the DOM -- it was a wrapper that visually hid the added image from the screen, and always the same wrapper was used to just add it once (not a separate one per every element). If we can just pass the image node, there's no reason at all for creating that `div`.

**Comment by spacedmonkey** [comment]
I had a quick look into this and this code needs refactoring now. It is very confusing to follow. I have created issue for this. https://github.com/google/web-stories-wp/issues/10025

For now, I am going back to simple fix, which means passing an id, which fixes the duplicate color issues, I was seeing. Longer term, I really want to refactor this code.

**Comment by swissspidy** [comment]
I'd like to properly fix and refactor this in the current/upcoming sprint.

**Comment by spacedmonkey** [comment]
Created issue https://github.com/google/web-stories-wp/issues/10099 and closing as larger refactor is needed.

**Pull request closed without merging**